### PR TITLE
Support scala 2.13.0-RC1

### PR DESCRIPTION
**Pull request by Philippus**
This is a follow-up of #72. I avoided the need to copy the whole `Shrink` file. Let me know if you like it like this or prefer another solution.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -84,19 +84,28 @@ lazy val compilationSettings = Seq(
...
+          Seq("-Yno-adapted-args", "-Ywarn-unused-import")
       }
     )
   , scalacOptions in (Compile,console) := Seq("-language:_", "-feature")
   , scalacOptions in (Test,console) := Seq("-language:_", "-feature")
+  , unmanagedSourceDirectories in Compile ++= {
+      (unmanagedSourceDirectories in Compile).value.map { dir =>
+        CrossVersion.partialVersion(scalaVersion.value) match {
+          case Some((2, 13)) => file(dir.getPath ++ "-2.13")
+          case _             => file(dir.getPath ++ "-2.10-2.12")
```
Comment (charleso) [inline build.sbt]: I'm very happy to go with whatever is the default convention across scala projects, but I noticed ScalaCheck used `2-13.+` and `2.13-` which I quite like for brevity. I'll defer to whatever you think is best here.

https://github.com/rickynils/scalacheck/tree/master/src/main

File: core/src/main/scala-2.10-2.12/hedgehog/core/XCompat.scala [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package hedgehog.core
+
+trait XCompat {
+  val StreamOrLazyList = scala.collection.immutable.Stream
```
Comment (charleso) [inline core/src/main/scala-2.10-2.12/hedgehog/core/XCompat.scala]: I don't love the name `StreamOrLazyList`, but I don't necessarily have a better suggestion. `StreamLike`? `Stream2`> I don't want to bikeshed it though so unless that gives you an idea let's just stick with for now.

**Comment by charleso** [comment]
Definitely better now. 💯 

I've left two nit comments. Let me know if you decide to action them (or not), and I'll merge after that.

File: build.sbt (hunk 1 above)
Comment (Philippus) [reply build.sbt]: :+1: It seems that way is suggested in https://github.com/scala/collection-strawman/wiki/FAQ#how-do-i-cross-build-my-project-against-scala-212-and-scala-213

File: core/src/main/scala-2.10-2.12/hedgehog/core/XCompat.scala (hunk 2 above)
Comment (Philippus) [reply core/src/main/scala-2.10-2.12/hedgehog/core/XCompat.scala]: I also don't like it, but don't have an alternative yet. We can leave it for now and when a better name pops up go with that.

**Pull request merged**